### ISSUE TITLE
MySQL SelectDecorator - subselect limit parameter not prefixed (fix included)

### ISSUE THREAD

**Issue by michalbundyra** [laminas/laminas-db#50]
- [x] I was not able to find an [open](https://github.com/zendframework/zend-db/issues?q=is%3Aopen) or [closed](https://github.com/zendframework/zend-db/issues?q=is%3Aclosed) issue matching what I'm seeing.
 - [x] This is not a question. (Questions should be asked on [slack](https://zendframework.slack.com/) ([Signup for Slack here](https://zendframework-slack.herokuapp.com/)) or our [forums](https://discourse.zendframework.com/).)

When you have a FROM clause that includes a query with a LIMIT in it, and you try to prepare the query, you get following error:

```
Fatal error: Uncaught PDOException: SQLSTATE[HY093]: Invalid parameter number: parameter was not defined in vendor/zendframework/zend-db/src/Adapter/Driver/Pdo/Statement.php:239
Stack trace:
#0 vendor/zendframework/zend-db/src/Adapter/Driver/Pdo/Statement.php(239): PDOStatement->execute()
```

The reason for this is because the parameter name for the subselect's limit is `:limit` while in the parameterContainer it is `:subselect1limit`.

This problem is rectified by modifying `line 55` of  `src/Sql/Platform/Mysql/SelectDecorator.php` from:

```php
return [$driver->formatParameterName('limit')];
```
to

```php
return [$driver->formatParameterName($paramPrefix . 'limit')];
```

---

Originally posted by @smartwebproducts at https://github.com/zendframework/zend-db/issues/318

**Comment by michalbundyra** [comment]
Looks like that was fixed by this PR #300 )

---

Originally posted by @tptrixtop at https://github.com/zendframework/zend-db/issues/318#issuecomment-389495074

**Comment by samsonasik** [comment]
it seems already fixed by merged PR https://github.com/zendframework/zend-db/pull/300

I am closing it.
